p2p/peer.go

```diff
@@ -214,97 +214,280 @@ type PeerUpdate struct {
 	Status PeerStatus
 }
 
-// peerStore manages information about peers. It is currently a bare-bones
-// in-memory store of peer addresses, and will be fleshed out later.
+// peerManager manages peer information, using a peerStore for underlying
+// storage. Its primary purpose is to determine which peers to connect to next,
+// make sure a peer only has a single active connection (either inbound or outbound),
+// and to avoid dialing the same peer in parallel goroutines.
 //
-// The main function of peerStore is currently to dispense peers to connect to
-// (via peerStore.Dispense), giving the caller exclusive "ownership" of that
-// peer until the peer is returned (via peerStore.Return). This is used to
-// schedule and synchronize peer dialing and accepting in the Router, e.g.
-// making sure we only have a single connection (in either direction) to peers.
-type peerStore struct {
-	mtx     sync.Mutex
-	peers   map[NodeID]*peerInfo
-	claimed map[NodeID]bool
+// For an outbound connection, the flow is as follows:
+// - DialNext: returns a peer address to dial, marking the peer as dialing.
+// - DialFailed: reports a dail failure, unmarking the peer as dialing.
+// - Dialed: successfully dialed, unmarking as dialing and marking as connected
+//   (or erroring if already connected).
+// - Ready: routing is up, broadcasts a PeerStatusUp peer update to subscribers.
+// - Disconnected: peer disconnects, unmarking as connected.
```

Comment:
Is there a peer status update sent in this event?

Reply:
Good catch, added a note of this.